src/mcts/search.cc

```diff
@@ -1214,7 +1214,15 @@ void SearchWorker::ExtendNode(Node* node) {
       return;
     }
 
-    if (history_.Last().GetRepetitions() >= 2) {
+    const auto repetitions = history_.Last().GetRepetitions();
+    const auto parent = node->GetParent();
+    // Treat two-fold repetitions as draw to help focus search on other moves,
+    // but avoid playing into a losing two-fold thinking it will end the game.
+    if (params_.GetTwoFoldDraw() && repetitions == 1 &&
+        parent != search_->root_node_) {
+      node->MakeTerminal(GameResult::DRAW, 0.0f, Node::Terminal::TwoFold);
```

Comment:
Not sure if M should be 0.0f to 2 fold draws.
I mean it shouldn't matter much since its a draw, but it isn't correct. It should be at least 4 more ply to get to the third rep?

Reply:
This sounds as if sending that position to the NN regularly and reporting the M might actually be the preferred way?